internal/store/store.go

```diff
@@ -0,0 +1,47 @@
+package store
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+type Store struct {
+	*pgxpool.Pool
+}
+
+func NewStore(ctx context.Context, connString string) (Store, error) {
+	dbpool, err := pgxpool.New(ctx, connString)
+	if err != nil {
+		return Store{}, err
+	}
+	return Store{dbpool}, nil
+}
+
+func (db Store) CreateUsersTable(ctx context.Context) error {
+	_, err := db.Exec(ctx,
+		`CREATE TABLE IF NOT EXISTS users (
+			id bigint GENERATED ALWAYS AS IDENTITY,
+			login text NOT NULL UNIQUE,
+			password text NOT NULL)`)
+	return err
+}
+
+func (db Store) AddUser(login string, hash []byte) (int, error) {
+	row := db.QueryRow(context.Background(), "INSERT INTO users (login, password) VALUES ($1, $2) RETURNING id", login, hash)
```

Comment:
Если мы используем контекст стоит передавать его извне. Так будет удобно отменять операцию при завершении приложения или управлять таймаутом.

```suggestion
func (db Store) AddUser(ctx  context.Context, login string, hash []byte) (int, error) {
	row := db.QueryRow(ctx, "INSERT INTO users (login, password) VALUES ($1, $2) RETURNING id", login, hash)
```